### PR TITLE
test: Fix test_demangle_stacks_symbol_map and run it under spidermonkey in CI. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: test/test_core.py [hunk 1]
```diff
@@ -7454,7 +7454,7 @@ def test_demangle_stacks_symbol_map(self):
     self.assertIsNotNone(full_aborter)
     self.assertIsNotNone(short_aborter)
     print('full:', full_aborter, 'short:', short_aborter)
-    if config.SPIDERMONKEY_ENGINE and os.path.exists(config.SPIDERMONKEY_ENGINE[0]):
+    if config.SPIDERMONKEY_ENGINE and config.SPIDERMONKEY_ENGINE == self.js_engines[0]:
```
Comment (kripken) [inline test/test_core.py:7457]: Why check if it's the first item in `js_engines`? Shouldn't we run the test if it is the second in the list?

Comment (sbc100) [reply test/test_core.py:7457]: I guess we could... I was thinking about that case when no `SPIDERMONKEY_ENGINE` at all was configured..  but I guess having `None` as  `self.js_engines[0]` would already be in undefined behaviour territory?

File: test/test_core.py [hunk 2]
```diff
@@ -7454,7 +7454,7 @@ def test_demangle_stacks_symbol_map(self):
     self.assertIsNotNone(full_aborter)
     self.assertIsNotNone(short_aborter)
     print('full:', full_aborter, 'short:', short_aborter)
-    if config.SPIDERMONKEY_ENGINE and os.path.exists(config.SPIDERMONKEY_ENGINE[0]):
+    if self.js_engines[0] == config.SPIDERMONKEY_ENGINE:
```
Comment (kripken) [inline test/test_core.py:7457]:
```suggestion
    if config.SPIDERMONKEY_ENGINE in self.js_engines:
```
Sorry if I wasn't clear before, this is what I meant. This should handle `js_engines = [NODE_JS, SPIDERMONKEY_ENGINE, ..]` etc. - which I think we want to support?

Comment (sbc100) [reply test/test_core.py:7457]: Yes that would make sense i think.. trying it out though it looks like this doesn't currently pass with current spider monkey.  I will see if I can make it pass and get this test running under spidermonkey in CI.

File: .circleci/config.yml [hunk 3]
```diff
@@ -623,7 +623,7 @@ jobs:
             echo "JSC_ENGINE = [os.path.expanduser('~/.jsvu/bin/spidermonkey')]" >> ~/emsdk/.emscripten
             echo "JS_ENGINES = [JSC_ENGINE]" >> ~/emsdk/.emscripten
```
Comment (kripken) [inline .circleci/config.yml:624]:
```suggestion
            echo "SPIDERMONKEY_ENGINE = [os.path.expanduser('~/.jsvu/bin/spidermonkey')]" >> ~/emsdk/.emscripten
            echo "SPIDERMONKEY_ENGINES = [JSC_ENGINE]" >> ~/emsdk/.emscripten
```

Comment (kripken) [reply .circleci/config.yml:624]: (existing, but I noticed it now)

**Review by kripken — COMMENTED** [review]
lgtm % comment (which could also be a followup)

File: .circleci/config.yml (hunk 3 above)
Comment (sbc100) [reply .circleci/config.yml:624]: Oh nice fine.. That is actually kind of key.

**Review by kripken — APPROVED** [review]

**Pull request merged**